dashboard/api.py

```diff
@@ -212,7 +248,43 @@ def get_info_for_course(user, course, user_enrollments, user_certificates):
         status = CourseStatus.OFFERED if next_run is not None else CourseStatus.NOT_OFFERED
         course_data['status'] = status
         if next_run is not None:
-            course_data['runs'].append(format_courserun_for_dashboard(next_run, status))
+            course_data['runs'].append(format_courserun_for_dashboard(next_run, status, position=index))
+            index += 1
+        # add the run of the status anyway if the next run is different from the one just added
+        if next_run is None or run_status.course_run.pk != next_run.pk:
+            course_data['runs'].append(
+                format_courserun_for_dashboard(
+                    run_status.course_run, CourseStatus.NOT_PASSED, position=index)
+            )
+            index += 1
+
+    # add all the other runs with status != NOT_ENROLLED
+    # the first one (or two in some cases) has been added with the logic before
+    for run_status in run_statuses:
```

Comment:
Putting the not-passed in a different list might be better so that we group together the different cases. At the end we can just concat the two lists together